### PR TITLE
Fixes #17049: Implement metadata for enums

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17049

### CODE REVIEW

File: rudder-lang/libs/oslib.rl [hunk 1]
```diff
@@ -41,13 +41,15 @@ items in bsd {
 items in linux {
   alpine_linux,
   arch_linux,
+  @cfe_name="debian"
```
Comment (amousset) [inline rudder-lang/libs/oslib.rl:44]: maybe `cfengine_name` to stay consistent with the name of the generator.

File: rudder-lang/libs/oslib.rl [hunk 2]
```diff
@@ -56,13 +58,15 @@ items in linux {
...
   centos,
+  @cfe_name="(redhat.!centos)"
   redhat,
   redhat_entreprise,
   scientific_linux,
   *
 }
 
 items in debian_family {
+  @cfe_name="(redhat.!ubuntu)"
```
Comment (amousset) [inline rudder-lang/libs/oslib.rl:69]: s/redhat/debian

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**